### DISCUSSION TITLE
How to enable server-side encryption for external storage?

### DISCUSSION THREAD

**Discussion by Wunderharke** [nextcloud/all-in-one#3937] [Questions] [answered]
As my server does not provide enough storage I'd lioke to use external S3 storage to store my data. For privacy reasons I'd like to have this data encrypted. This is normally possible via server-side encryption but this options seems to be disabled (probably for backups / fulltext-search to work).
Is there any approach to get this solves? I dont care that nextcloud owns the keys and would even elasticsearch to be able to index those files, I just want them to be encrpted-at-rest somehow.

**Comment by szaimen** [comment, marked as answer]
Hi, please ask this question at https://help.nextcloud.com. Thanks!
